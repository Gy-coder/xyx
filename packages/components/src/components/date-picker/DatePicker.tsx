import {
  ForwardRefRenderFunction,
  useCallback,
  useEffect,
  useMemo,
  useRef,
  useState,
} from "react";
import classnames from "classnames";
import Input from "../input";
import InternalIcon from "../_interal/internal_icon/InternalIcon";
import useClickOutSide from "../../hooks/useClickOutSide";
import useMergeState from "../../hooks/useMergeState";
import Dayjs from "../../utils/dayjs";
import DatePanel from "./DatePanel";
import type { DatePickerProps, modeType, valueType } from "./interface";
import "./index.scss";
import MonthPanel from "./MonthPanel";

const DatePicker: ForwardRefRenderFunction<any, DatePickerProps> = (
  props,
  ref
) => {
  const {
    defaultValue,
    value,
    onChange,
    allowClear = false,
    placeholder = "点击输入日期",
  } = props;
  const [stateValue, _setStateValue, isControlled] = useMergeState<valueType>(
    undefined,
    {
      defaultValue,
      value,
    }
  );
  const [visiblePanel, setVisiblePanel] = useState<boolean>(false);
  const [mode, setMode] = useState<modeType>("date");
  const [visibleValue, setVisibleValue] = useState<Dayjs>(
    new Dayjs(stateValue)
  );
  const innerValue: Dayjs | undefined = useMemo(
    () => (stateValue ? new Dayjs(stateValue) : undefined),
    [stateValue]
  );
  const componentRef = useRef<HTMLDivElement>(null);
  const openPanel = useCallback(() => {
    setMode("date");
    setVisibleValue(innerValue || new Dayjs());
    setVisiblePanel(true);
  }, [innerValue]);
  const closePanel = useCallback(() => {
    setVisiblePanel(false);
  }, []);
  const setStateValue = (newValue?: Date, valueString?: string) => {
    if (!isControlled) _setStateValue(newValue);
    onChange?.(newValue, valueString);
  };
  const handleClear = () => {
    setStateValue(undefined, undefined);
  };
  const handleChangeMode = (newMode: modeType) => setMode(newMode);
  useClickOutSide(componentRef, () => {
    closePanel();
  });
  useEffect(() => setVisibleValue(innerValue || new Dayjs()), [innerValue]);
  const renderPanel = () => {
    switch (mode) {
      case "date":
        return (
          <DatePanel
            visibleValue={visibleValue}
            onChangeVisibleValue={setVisibleValue}
            onChangeValue={setStateValue}
            closePanel={closePanel}
            onChangeMode={handleChangeMode}
            innerValue={innerValue}
          />
        );
      case "month":
        return <MonthPanel />;
      case "year":
        return <div>year</div>;
    }
  };
  return (
    <div className={classnames("g-datepicker")} ref={componentRef}>
      {innerValue?.format()}
      <Input
        suffix={<InternalIcon name="icon-date" />}
        width={200}
        readOnly
        onClick={openPanel}
        value={innerValue ? innerValue.format() : ""}
        clearable={allowClear}
        onClickClearButton={handleClear}
        placeholder={placeholder}
      />
      {visiblePanel && (
        <div className={classnames("g-datepicker-panel-wrapper")}>
          {renderPanel()}
        </div>
      )}
    </div>
  );
};

export default DatePicker;
