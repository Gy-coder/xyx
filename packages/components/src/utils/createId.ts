let id = 0;

export default function createId(){
    return id++
}