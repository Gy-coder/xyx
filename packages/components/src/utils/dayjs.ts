type DayUnit = "year"
    | "month"
    | "day"
    | "hour"
    | "minute"
    | "second"
    | "millisecond";

export default class Dayjs {
    public date: Date

    constructor(date?: number | string | Date) {
        this.date = date ? new Date(date) : new Date()
    }

    get year() {
        return this.date.getFullYear()
    }

    get month() {
        return this.date.getMonth() + 1
    }

    get week() {
        return this.date.getDay()
    }

    get day() {
        return this.date.getDate()
    }

    get hour() {
        return this.date.getHours()
    }

    get minute() {
        return this.date.getMinutes()
    }

    get second() {
        return this.date.getSeconds()
    }

    get millisecond() {
        return this.date.getMilliseconds()
    }

    get raw() {
        return this.date
    }

    public firstDayOfMonth() {
        return new Dayjs(
            new Date(
                this.year,
                this.month - 1,
                1,
                0,
                0,
                0)
        )
    }

    public add(amount: number, unit: DayUnit) {
        let date = new Date(this.date.getTime());
        switch (unit) {
            case 'year':
                const currentDate = date.getDate()
                date.setDate(1)
                date.setFullYear(date.getFullYear() + amount)
                const targetDate = new Date(
                    date.getFullYear(),
                    date.getMonth() + 1,
                    0,
                    0,
                    0,
                    0,
                ).getDate()
                date.setDate(Math.min(currentDate, targetDate))
                break;
            case 'month':
                const d = date.getDate()
                date.setDate(1)
                date.setMonth(date.getMonth() + amount);
                const d2 = new Date(date.getFullYear(), date.getMonth() + 1, 0, 0, 0, 0).getDate()
                date.setDate(Math.min(d, d2))
                break;
            case 'day':
                date.setDate(date.getDate() + amount);
                break;
            case 'hour':
                date.setHours(date.getHours() + amount);
                break;
            case 'minute':
                date.setMinutes(date.getMinutes() + amount);
                break;
            case 'second':
                date.setSeconds(date.getSeconds() + amount);
                break;
            case 'millisecond':
                date.setMilliseconds(date.getMilliseconds() + amount);
                break;

            default:
                throw new Error('Time.add: unknown unit');
        }
        return new Dayjs(date)
    }

    public set(amount: number,unit: DayUnit){
        const date = new Date(this.date.getTime())
        switch (unit){
            case "year":
                date.setFullYear(amount)
                break
            case "month":
                date.setMonth(amount - 1)
                break
            case "day":
                date.setDate(amount)
                break
            case "hour":
                date.setHours(amount)
                break
            case "second":
                date.setSeconds(amount)
                break
            case "millisecond":
                date.setMilliseconds(amount)
                break
            default:
                throw new Error("Time.add: unknown unit")
        }
        return new Dayjs(date)
    }

    isSameDay(day: Dayjs){
        return day.year === this.year && day.month === this.month && day.day === this.day
    }

    isSameMonth(day: Dayjs){
        return day.year === this.year && day.month === this.month
    }

    //@ts-ignore
    format(pattern?: string = 'YYYY-MM-DD') {
        // 目前支持的格式有 YYYY MM DD HH mm ss SSS
        return pattern.replace(/YYYY/g, this.year.toString())
            .replace(/MM/, this.month.toString().padStart(2, '0'))
            .replace(/DD/, this.day.toString().padStart(2, '0'))
            .replace(/HH/, this.hour.toString().padStart(2, '0'))
            .replace(/mm/, this.minute.toString().padStart(2, '0'))
            .replace(/ss/, this.second.toString().padStart(2, '0'))
            .replace(/SSS/, this.millisecond.toString().padStart(3, '0'))
    }
}
