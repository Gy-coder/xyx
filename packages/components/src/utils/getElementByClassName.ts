export function getElementByClassName(container = document, classname: string) {
  return container.getElementsByClassName(classname)[0];
}
