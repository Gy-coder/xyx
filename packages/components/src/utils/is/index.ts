export { default as isUndefined } from "./isUndefined";
export { default as isFunction } from "./isFunction";
