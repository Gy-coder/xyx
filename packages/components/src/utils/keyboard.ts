export const keyboard = {
  Enter: "Enter",
};
