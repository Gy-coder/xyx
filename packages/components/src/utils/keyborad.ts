export const keyborad = {
  Enter: "Enter",
};
