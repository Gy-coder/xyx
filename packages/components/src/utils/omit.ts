export default function omit<T = any>(obj: Record<string, T>, paths: string[]) {
  const res: Record<string, T> = { ...obj },
    set = new Set(paths);
  for (let key in obj) {
    if (set.has(key)) delete res[key];
  }
  return res;
}
