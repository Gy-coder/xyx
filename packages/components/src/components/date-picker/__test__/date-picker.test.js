import { fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import DatePicker from "../index";
import { getElementByClassName } from "../../../utils/getElementByClassName";
import Dayjs from "../../../utils/dayjs";

describe("test DatePicker", () => {
  it("should Render Correctly", () => {
    const { container, asFragment } = render(<DatePicker />);
    const el = getElementByClassName(container, "g-datepicker");
    expect(el).toBeVisible();
    expect(asFragment()).toMatchSnapshot();
  });
  it("click can open picker", () => {
    const { container, asFragment } = render(<DatePicker placeholder="1234" />);
    const input = screen.getByPlaceholderText("1234");
    expect(input).toBeVisible();
    fireEvent.click(input);
    const el = getElementByClassName(container, "g-datepicker-panel-wrapper");
    expect(el).toBeVisible();
    expect(asFragment()).toMatchSnapshot();
    fireEvent.click(document);
    expect(el).not.toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
  it("onchange can work", () => {
    const jestFn = jest.fn();
    const { asFragment } = render(
      <DatePicker
        placeholder="1234"
        onChange={jestFn}
        defaultValue="2023-01-21"
      />
    );
    const input = screen.getByPlaceholderText("1234");
    fireEvent.click(input);
    const item = screen.getAllByText("1")[0];
    fireEvent.click(item);
    expect(jestFn).toBeCalledWith(new Date(2023, 0, 1, 0, 0, 0), "2023-01-01");
    expect(asFragment()).toMatchSnapshot();
  });
  it("clear", () => {
    const { container } = render(
      <DatePicker placeholder="1234" defaultValue="2023-01-21" allowClear />
    );
    const input = screen.getByDisplayValue("2023-01-21");
    const clear = getElementByClassName(container, "g-input-clear");
    expect(clear).toBeVisible();
    expect(input.value).toBe("2023-01-21");
    fireEvent.click(clear);
    expect(clear).not.toBeVisible();
    expect(input.value).toBe("");
  });
  it("click double left button", () => {
    const { container } = render(<DatePicker placeholder="1234" />);
    const input = screen.getByPlaceholderText("1234");
    fireEvent.click(input);
    const arrow = container.getElementsByClassName(
      "g-datepicker-day-picker-header-arrow"
    )[0];
    fireEvent.click(arrow);
    const el = container.querySelectorAll(
      ".g-datepicker-day-picker-header-day-and-month span"
    )[0];
    expect(el.innerHTML).toBe(new Dayjs().year - 1 + "年");
  });
  it("click left button", () => {
    const { container } = render(<DatePicker placeholder="1234" />);
    const input = screen.getByPlaceholderText("1234");
    fireEvent.click(input);
    const arrow = container.getElementsByClassName(
      "g-datepicker-day-picker-header-arrow"
    )[1];
    fireEvent.click(arrow);
    const els = container.querySelectorAll(
      ".g-datepicker-day-picker-header-day-and-month span"
    );
    const month = parseInt(els[1].innerHTML);
    let year = new Dayjs().year;
    if (month > new Dayjs().month) {
      year--;
      // eslint-disable-next-line jest/no-conditional-expect
      expect(els[0].innerHTML).toBe(year + "年");
      // eslint-disable-next-line jest/no-conditional-expect
      expect(els[1].innerHTML).toBe(12 + "月");
    } else {
      // eslint-disable-next-line jest/no-conditional-expect
      expect(els[1].innerHTML).toBe(new Dayjs().month - 1 + "月");
    }
  });
  it("test format", () => {
    const jestFn = jest.fn();
    const { container } = render(
      <DatePicker format="YYYY/MM/DD" onChange={jestFn} />
    );
    const input = getElementByClassName(container, "g-input-origin");
    fireEvent.click(input);
    const footer = screen.getByText("今 天");
    fireEvent.click(footer);
    const year = new Date().getFullYear(),
      month = (new Date().getMonth() + 1).toString().padStart(2, "0"),
      day = new Date().getDate().toString().padStart(2, "0");
    expect(input.value).toBe(`${year}/${month}/${day}`);
    // expect(jestFn).toBeCalledWith(new Date(), `${year}/${month}/${day}`);
    expect(footer).not.toBeInTheDocument();
  });
  it("test month picker", () => {
    const jestFn = jest.fn();
    const { container, asFragment } = render(
      <DatePicker picker="month" onChange={jestFn} />
    );
    const input = getElementByClassName(container, "g-input-origin");
    fireEvent.click(input);
    const el = getElementByClassName(
      container,
      "g-datepicker-panel-header-middle"
    );
    expect(el).toBeVisible();
    expect(el.textContent).toBe(`${new Date().getFullYear()}年`);
    expect(asFragment()).toMatchSnapshot();
    const item = screen.getByText("十二月");
    fireEvent.click(item);
    // expect(jestFn).toBeCalledWith(new Date(), `${year}/${12}`);
    expect(input.value).toBe(`${new Date().getFullYear()}-12`);
    expect(el).not.toBeInTheDocument();
  });
});
