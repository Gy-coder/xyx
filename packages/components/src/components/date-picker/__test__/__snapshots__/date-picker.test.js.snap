// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test DatePicker click can open picker 1`] = `
<DocumentFragment>
  <div
    class="g-datepicker"
  >
    <div
      class="g-input g-input-middle"
    >
      <input
        class="g-input-origin"
        placeholder="1234"
        readonly=""
        type="text"
        value=""
        width="200"
      />
      <span
        class="g-input-suffix"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="#icon-date"
          />
        </svg>
      </span>
    </div>
    <div
      class="g-datepicker-panel-wrapper"
    >
      <div
        class="g-datepicker-day-picker"
      >
        <header
          class="g-datepicker-day-picker-header"
        >
          <div>
            <span
              class="g-datepicker-day-picker-header-arrow"
            >
              <svg
                class="icon"
              >
                <use
                  xlink:href="#icon-arrow-double-left"
                />
              </svg>
            </span>
            <span
              class="g-datepicker-day-picker-header-arrow"
            >
              <svg
                class="icon"
              >
                <use
                  xlink:href="#icon-arrow-left"
                />
              </svg>
            </span>
          </div>
          <div
            class="g-datepicker-day-picker-header-day-and-month"
          >
            <span>
              2023年
            </span>
            <span>
              2月
            </span>
          </div>
          <div>
            <span
              class="g-datepicker-day-picker-header-arrow"
            >
              <svg
                class="icon"
              >
                <use
                  xlink:href="#icon-arrow-right"
                />
              </svg>
            </span>
            <span
              class="g-datepicker-day-picker-header-arrow"
            >
              <svg
                class="icon"
              >
                <use
                  xlink:href="#icon-arrow-double-right"
                />
              </svg>
            </span>
          </div>
        </header>
        <main
          class="g-datepicker-day-picker-content"
        >
          <span
            class="g-datepicker-day-picker-row"
          >
            <span
              class="g-datepicker-day-picker-title"
            >
              日
            </span>
            <span
              class="g-datepicker-day-picker-title"
            >
              一
            </span>
            <span
              class="g-datepicker-day-picker-title"
            >
              二
            </span>
            <span
              class="g-datepicker-day-picker-title"
            >
              三
            </span>
            <span
              class="g-datepicker-day-picker-title"
            >
              四
            </span>
            <span
              class="g-datepicker-day-picker-title"
            >
              五
            </span>
            <span
              class="g-datepicker-day-picker-title"
            >
              六
            </span>
          </span>
          <span
            class="g-datepicker-day-picker-row"
          >
            <div>
              <span
                class="g-datepicker-day-picker-item g-datepicker-day-picker-item-not-in-this-month"
              >
                29
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item g-datepicker-day-picker-item-not-in-this-month"
              >
                30
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item g-datepicker-day-picker-item-not-in-this-month"
              >
                31
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item"
              >
                1
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item"
              >
                2
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item"
              >
                3
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item"
              >
                4
              </span>
            </div>
          </span>
          <span
            class="g-datepicker-day-picker-row"
          >
            <div>
              <span
                class="g-datepicker-day-picker-item"
              >
                5
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item"
              >
                6
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item"
              >
                7
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item"
              >
                8
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item"
              >
                9
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item"
              >
                10
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item"
              >
                11
              </span>
            </div>
          </span>
          <span
            class="g-datepicker-day-picker-row"
          >
            <div>
              <span
                class="g-datepicker-day-picker-item"
              >
                12
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item"
              >
                13
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item"
              >
                14
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item"
              >
                15
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item"
              >
                16
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item"
              >
                17
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item"
              >
                18
              </span>
            </div>
          </span>
          <span
            class="g-datepicker-day-picker-row"
          >
            <div>
              <span
                class="g-datepicker-day-picker-item"
              >
                19
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item"
              >
                20
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item"
              >
                21
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item"
              >
                22
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item"
              >
                23
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item"
              >
                24
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item"
              >
                25
              </span>
            </div>
          </span>
          <span
            class="g-datepicker-day-picker-row"
          >
            <div>
              <span
                class="g-datepicker-day-picker-item"
              >
                26
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item"
              >
                27
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item"
              >
                28
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item g-datepicker-day-picker-item-not-in-this-month"
              >
                1
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item g-datepicker-day-picker-item-not-in-this-month"
              >
                2
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item g-datepicker-day-picker-item-not-in-this-month"
              >
                3
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item g-datepicker-day-picker-item-not-in-this-month"
              >
                4
              </span>
            </div>
          </span>
          <span
            class="g-datepicker-day-picker-row"
          >
            <div>
              <span
                class="g-datepicker-day-picker-item g-datepicker-day-picker-item-not-in-this-month"
              >
                5
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item g-datepicker-day-picker-item-not-in-this-month"
              >
                6
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item g-datepicker-day-picker-item-not-in-this-month"
              >
                7
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item g-datepicker-day-picker-item-not-in-this-month"
              >
                8
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item g-datepicker-day-picker-item-not-in-this-month"
              >
                9
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item g-datepicker-day-picker-item-not-in-this-month"
              >
                10
              </span>
            </div>
            <div>
              <span
                class="g-datepicker-day-picker-item g-datepicker-day-picker-item-not-in-this-month"
              >
                11
              </span>
            </div>
          </span>
        </main>
        <footer
          class="g-datepicker-day-picker-footer"
        >
          <span>
            今 天
          </span>
        </footer>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`test DatePicker click can open picker 2`] = `
<DocumentFragment>
  <div
    class="g-datepicker"
  >
    <div
      class="g-input g-input-middle"
    >
      <input
        class="g-input-origin"
        placeholder="1234"
        readonly=""
        type="text"
        value=""
        width="200"
      />
      <span
        class="g-input-suffix"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="#icon-date"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`test DatePicker onchange can work 1`] = `
<DocumentFragment>
  <div
    class="g-datepicker"
  >
    <div
      class="g-input g-input-middle"
    >
      <input
        class="g-input-origin"
        placeholder="1234"
        readonly=""
        type="text"
        value="2023-01-01"
        width="200"
      />
      <span
        class="g-input-suffix"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="#icon-date"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`test DatePicker should Render Correctly 1`] = `
<DocumentFragment>
  <div
    class="g-datepicker"
  >
    <div
      class="g-input g-input-middle"
    >
      <input
        class="g-input-origin"
        placeholder="点击输入日期"
        readonly=""
        type="text"
        value=""
        width="200"
      />
      <span
        class="g-input-suffix"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="#icon-date"
          />
        </svg>
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`test DatePicker test month picker 1`] = `
<DocumentFragment>
  <div
    class="g-datepicker"
  >
    <div
      class="g-input g-input-middle"
    >
      <input
        class="g-input-origin"
        placeholder="点击输入日期"
        readonly=""
        type="text"
        value=""
        width="200"
      />
      <span
        class="g-input-suffix"
      >
        <svg
          class="icon"
        >
          <use
            xlink:href="#icon-date"
          />
        </svg>
      </span>
    </div>
    <div
      class="g-datepicker-panel-wrapper"
    >
      <div
        class="g-datepicker-panel"
      >
        <header
          class="g-datepicker-panel-header"
        >
          <span
            class="g-datepicker-panel-header-arrow"
          >
            <svg
              class="icon"
            >
              <use
                xlink:href="#icon-arrow-double-left"
              />
            </svg>
          </span>
          <span
            class="g-datepicker-panel-header-middle"
          >
            2023年
          </span>
          <span
            class="g-datepicker-panel-header-arrow"
          >
            <svg
              class="icon"
            >
              <use
                xlink:href="#icon-arrow-double-right"
              />
            </svg>
          </span>
        </header>
        <main
          class="g-datepicker-panel-content"
        >
          <div
            class="g-datepicker-panel-content-row"
          >
            <div>
              <span
                class=""
              >
                  一月
              </span>
            </div>
            <div>
              <span
                class=""
              >
                  二月
              </span>
            </div>
            <div>
              <span
                class=""
              >
                  三月
              </span>
            </div>
          </div>
          <div
            class="g-datepicker-panel-content-row"
          >
            <div>
              <span
                class=""
              >
                  四月
              </span>
            </div>
            <div>
              <span
                class=""
              >
                  五月
              </span>
            </div>
            <div>
              <span
                class=""
              >
                  六月
              </span>
            </div>
          </div>
          <div
            class="g-datepicker-panel-content-row"
          >
            <div>
              <span
                class=""
              >
                  七月
              </span>
            </div>
            <div>
              <span
                class=""
              >
                  八月
              </span>
            </div>
            <div>
              <span
                class=""
              >
                  九月
              </span>
            </div>
          </div>
          <div
            class="g-datepicker-panel-content-row"
          >
            <div>
              <span
                class=""
              >
                  十月
              </span>
            </div>
            <div>
              <span
                class=""
              >
                  十一月
              </span>
            </div>
            <div>
              <span
                class=""
              >
                  十二月
              </span>
            </div>
          </div>
        </main>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
